lib/puppet/type/file/source.rb

```diff
@@ -274,17 +262,17 @@ def copy_source_value(metadata_method)
     end
 
     def each_chunk_from
-      if Puppet[:default_file_terminus] == :file_server
-        yield content
+      if Puppet[:default_file_terminus] == :file_server && scheme == 'puppet' && (uri.host.nil? || uri.host.empty?)
+        chunk_file_from_disk(metadata.path) { |chunk| yield chunk }
       elsif local?
-        chunk_file_from_disk { |chunk| yield chunk }
+        chunk_file_from_disk(full_path) { |chunk| yield chunk }
       else
         chunk_file_from_source { |chunk| yield chunk }
       end
     end
 
-    def chunk_file_from_disk
-      File.open(full_path, "rb") do |src|
+    def chunk_file_from_disk(local_path)
```

Comment:
This is technically not a backwards compatible change, but since this is targeted at master, I think it should be fine, but I wanted to double check on that

Reply:
In this case, it's ok because there's a `private` keyword up above.

Reply:
Oh cool, I did not notice that